### PR TITLE
Oli

### LINKED ISSUE #2: Update the `/friend` endpoint to add a new friend to the database
# Description

What we are build is a REST API, we need to have a endpoint that can receive  a POST request with some data `NewFriend` and insert it into the database 

### Notes

- This api uses `FastApi` framework https://fastapi.tiangolo.com/ 
- The creation of the endpoint is already taken care of you just need to fill in the function.
- For using the database we use a package called `sqlalchemy`  https://docs.sqlalchemy.org/en/20/orm/session_basics.html#adding-new-or-existing-items

### PR DESCRIPTION
**Pull request by AndyM10**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -37,4 +39,27 @@ def read_root(db: db_dependency):
 
 @app.post("/friend")
 def add_friend(friend: NewFriend, db: db_dependency):
-    return friend
+    db_friend = models.Friend(name=friend.name, location=friend.location, friends=friend.friends)
+    db.add(db_friend)
+    db.commit()
+    return 'Friend Added'
+
+@app.delete('/FRIENDS/{friend_id}')
```
Comment (AndyM10) [inline app/main.py]: We should make these route handlers lowercase for consistency, e.g (`/friends/{friend_id}`)

File: app/main.py [hunk 2]
```diff
@@ -37,4 +39,27 @@ def read_root(db: db_dependency):
...
+    for x in db.scalars(db_query):
+        db.delete(x)
+    db.commit()
+    return "Friend Removed"
+
+@app.put('/FRIENDS/{update}')
+def update_friend(friend: NewFriend, friend_id: int, db: db_dependency):
+    new_name = friend.name
+    new_location = friend.location
+    new_friends = friend.friends
```
Comment (AndyM10) [inline app/main.py]: There is a way to write this on one line with a method called deconstruction :)

File: app/main.py [hunk 3]
```diff
@@ -37,4 +39,27 @@ def read_root(db: db_dependency):
...
+    db_query = select(models.Friend).where(models.Friend.id == friend_id)
```
Comment (AndyM10) [inline app/main.py]: I wonder if we can use the `update` function here to write this in a more idiomatic way?

Comment (Oliverbell97) [reply app/main.py]: Have you got any documentation on the .update() cause I can only see update as a parameter on the FastAPI notes?

Comment (Oliverbell97) [reply app/main.py]: Our database is an array instead of a dictionary, would it work better as a dictionary?

Comment (AndyM10) [reply app/main.py]: https://docs.sqlalchemy.org/en/20/core/dml.html#sqlalchemy.sql.expression.Update.where

Comment (AndyM10) [reply app/main.py]: I got it working :)

Comment (Oliverbell97) [reply app/main.py]: Could you show me how you got it? I keep trying to use .update() on either db or models.Friend but its saying it doesnt have that function

**Pull request merged**